### PR TITLE
#45 Nested test suite support

### PR DESCRIPTION
**Pull request by jamesagardner**
Code cleanup and refactor to support nested test suites.

### CODE REVIEW

**Comment by jamesagardner** [comment]
Going to re-create pull request targeting the develop branch.

**Comment by dmcassel** [comment]
I regularly set up a "lib" directory in a test suite that holds a lib.xqy, containing variables and functions used across my tests. If I'm reading correctly, that won't work anymore. I guess I can move lib.xqy into test-data, though, and that would prevent it from being seen as a test, correct? This will be a breaking change, which should be indicated in the version number. 

Can you please confirm that you've run the project's unit tests in both UI and from gradle?

File: marklogic-unit-test-modules/src/main/ml-modules/root/test/test-helper.xqy [hunk 1]
```diff
@@ -665,30 +663,30 @@ declare function helper:log($items as item()*)
...
-      fn:concat($root, 'test/', $file-type, '/', ($suite, '')[1])),
-      <options xmlns="xdmp:eval"><database>{$database}</database></options>)
+  (: Add trailing '/' if missing :)
+  let $path := fn:replace($path, "([^/])$", "$1/")
+  return
+    if ($database = 0) then
+      let $directory-separator := if (xdmp:platform() eq "winnt") then "\\" else "/"
+      return helper:list-from-filesystem(fn:replace($path, "(/|\\)", $directory-separator))
+    else
+      xdmp:eval(
```
Comment (hansenmc) [inline marklogic-unit-test-modules/src/main/ml-modules/root/test/test-helper.xqy]: We should consider replacing `xdmp:eval()` with `xdmp:invoke-function()`. Maybe not in this set of commits, to minimize changes during the refactor, but something to consider for future work.  https://www.marklogic.com/blog/first-class-functions/

**Comment by hansenmc** [comment]
@dmcassel `test-lib` and `test-data` are already in the set of `$directories-to-ignore`. 
https://github.com/marklogic-community/marklogic-unit-test/pull/54/commits/402759e1f02c6ca2feae88c595cd376dab86d26f#diff-0ca38b15b5b7fbdf085da371cea40f14L28 

If you were to rename your `lib` to `test-lib`, that should work and would match the expected conventions.

**Comment by jamesagardner** [comment]
@dmcassel I confirmed that both the gradle test target as well as the :8090/test/default.xqy UI pass all tests with the above code commit.

Based on your feedback I decided to protect /lib directories instead of /test-lib.  It was my original thought as well, but saw that instead of protecting /data, we protected /test-data, so I thought there was a convention.

I do note that gradle targets need to be ran from the marklogic-unit-test-client sub project or ml-gradle seems to miss deployment artifacts when ran from the root project.

**Pull request merged**